atmel-samd/tick.c

```diff
@@ -29,18 +45,23 @@ void timer_tick(const struct timer_task *const timer_task) {
 }
 
 void tick_init() {
-    #ifdef SAMD21
-    _pm_enable_bus_clock(PM_BUS_APBC, TC5);
-    #endif
-    // _gclk_enable_channel(TC5_GCLK_ID, GCLK_SOURCE_DFLL48M);
-
-    // timer_init(&ms_timer, TC5, _tc_get_timer());
+    uint32_t ticks_per_ms = common_hal_mcu_processor_get_frequency() / 1000;
+    SysTick->LOAD = ((ticks_per_ms - 1) << SysTick_LOAD_RELOAD_Pos);
```

Comment:
You could do `SysTick_Config(ticks_per_ms)` to replace lines 49-53.

I remembered what Dean told me about CMSIS and thought there might be a function to do this.